ios/RCTConvert+Notification.m

```diff
@@ -93,37 +93,59 @@ @implementation RCTConvert (UNNotificationRequest)
-    
+
     BOOL isSilent = [RCTConvert BOOL:details[@"isSilent"]];
     NSString* identifier = [RCTConvert NSString:details[@"id"]];
-    
-    
+
     UNMutableNotificationContent* content = [[UNMutableNotificationContent alloc] init];
-    content.title= [RCTConvert NSString:details[@"title"]];
-    content.subtitle= [RCTConvert NSString:details[@"subtitle"]];
-    content.body =[RCTConvert NSString:details[@"body"]];
-    content.badge = [RCTConvert NSNumber:details[@"badge"]];
+    content.title              = [RCTConvert NSString:details[@"title"]];
+    content.subtitle           = [RCTConvert NSString:details[@"subtitle"]];
+    content.body               = [RCTConvert NSString:details[@"body"]];
+    content.badge              = [RCTConvert NSNumber:details[@"badge"]];
     content.categoryIdentifier = [RCTConvert NSString:details[@"category"]];
 
     NSString* threadIdentifier = [RCTConvert NSString:details[@"threadId"]];
     if (threadIdentifier){
         content.threadIdentifier = threadIdentifier;
     }
 
+    NSDictionary<NSString *, id> *userDateComps = [RCTConvert NSDictionary:details[@"repeatsComponent"]];
+    BOOL month      = [RCTConvert BOOL:userDateComps[@"month"]];
+    BOOL day        = [RCTConvert BOOL:userDateComps[@"day"]];
+    BOOL hour       = [RCTConvert BOOL:userDateComps[@"hour"]];
+    BOOL minute     = [RCTConvert BOOL:userDateComps[@"minute"]];
+    BOOL second     = [RCTConvert BOOL:userDateComps[@"second"]];
+    BOOL nanosecond = [RCTConvert BOOL:userDateComps[@"nanosecond"]];
+
     content.userInfo = [RCTConvert NSDictionary:details[@"userInfo"]];
     if (!isSilent) {
-      content.sound = [RCTConvert NSString:details[@"sound"]] ? [UNNotificationSound soundNamed:[RCTConvert NSString:details[@"sound"]]] : [UNNotificationSound defaultSound];
+      content.sound = [RCTConvert NSString:details[@"sound"]]
+        ? [UNNotificationSound soundNamed:[RCTConvert NSString:details[@"sound"]]]
+        : [UNNotificationSound defaultSound];
     }
 
     NSDate* fireDate = [RCTConvert NSDate:details[@"fireDate"]];
     BOOL repeats = [RCTConvert BOOL:details[@"repeats"]];
-    NSDateComponents *triggerDate = fireDate ? [[NSCalendar currentCalendar]
-                                                components:NSCalendarUnitYear +
-                                                NSCalendarUnitMonth + NSCalendarUnitDay +
-                                                NSCalendarUnitHour + NSCalendarUnitMinute +
-                                                NSCalendarUnitSecond + NSCalendarUnitTimeZone
-                                                fromDate:fireDate] : nil;
-    
+    NSCalendarUnit defaultDateComponents =
+        NSCalendarUnitYear |
+        NSCalendarUnitMonth |
+        NSCalendarUnitDay |
+        NSCalendarUnitHour |
+        NSCalendarUnitMinute |
+        NSCalendarUnitSecond;
+    NSCalendarUnit repeatDateComponents =
```

Comment:
Suggest addition of NSCalendarUnitYear and NSCalendarUnitWeekday.
Suggest removal of NSCalendarUnitNanosecond.